Zend/zend_builtin_functions.c

```diff
@@ -69,6 +69,29 @@ zend_result zend_startup_builtin_functions(void) /* {{{ */
 }
 /* }}} */
 
+ZEND_FUNCTION(exit)
+{
+	zend_string *str = NULL;
+	zend_long code = 0;
+
+	ZEND_PARSE_PARAMETERS_START(0, 1)
+		Z_PARAM_OPTIONAL
+		Z_PARAM_STR_OR_LONG(str, code)
+	ZEND_PARSE_PARAMETERS_END();
+
+	if (str) {
+		size_t len = ZSTR_LEN(str);
+		if (len != 0) {
+			zend_write(ZSTR_VAL(str), len);
+		}
+	} else {
+		EG(exit_status) = code;
+	}
+
+	ZEND_ASSERT(!EG(exception));
```

Comment:
This assert can fail.

```php
ob_start(function ($text) {
    fwrite(STDOUT, "Handler: " . $text);
    throw new Exception('test');
}, chunk_size: 10);

exit("Hello world!\n");
```

Reply:
Indeed, I didn't think of the output buffer, added a test